apps/openmw/mwmechanics/weaponpriority.cpp

```diff
@@ -90,10 +92,21 @@ namespace MWMechanics
         }
         else if (weapon->mData.mType == ESM::Weapon::MarksmanCrossbow)
         {
-            if (boltRating <= 0.f)
-                rating = 0.f;
-            else
-                rating += boltRating;
+            int ammotype = MWMechanics::getWeaponType(weapon->mData.mType)->mAmmoType;
```

Comment:
Special handling of crossbow ammo type seems odd. Maybe instead of only checking if the weapon has crossbow type or bow type *any* ranged weapon should be checked for whether it has arrow ammo or bolt ammo? Basically putting this whole crossbow logic into the else path of "weapclass != ESM::WeaponType::Ranged" check.